### PR TITLE
[WIP][SPARK-31089][SQL] Add back ImageSchema.readImages in Spark 3.0

### PR DESCRIPTION
**Pull request by WeichenXu123**
### What changes were proposed in this pull request?

Add back ImageSchema.readImages in Spark 3.0

### CODE REVIEW

**Review by srowen — COMMENTED** [review]
Hm, this is one of those things I think is worth removing. I don't think this was widely used, but I don't have hard data for that. format("image") is more standard and all I have seen used, at least.

**Comment by gatorsmile** [comment]
cc @mengxr WDYT?

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Hi, @WeichenXu123 . Thank you for making a PR with test cases together in one piece.

However, I don't think this is a `MINOR` because we should not add back a **public** API silently like this. Please file a JIRA officially if you want. And, let's hold on this PR until we have a vote.

BTW, I agree with @srowen 's opinion.

File: mllib/src/main/scala/org/apache/spark/ml/image/ImageSchema.scala [hunk 1]
```diff
@@ -188,4 +189,76 @@ object ImageSchema {
...
+   * there may be a race condition where one job overwrites the hadoop configs of another.
+   * @note If sample ratio is less than 1, sampling uses a PathFilter that is efficient but
+   * potentially non-deterministic.
+   *
+   * @param path Path to the image directory
+   * @return DataFrame with a single column "image" of images;
+   *         see ImageSchema for the details
+   */
+  @deprecated("use `spark.read.format(\"image\").load(path)` and this `readImages` will be " +
+    "removed in 3.1.0.", "2.4.0")
```
Comment (dongjoon-hyun) [inline mllib/src/main/scala/org/apache/spark/ml/image/ImageSchema.scala]: In any way, this is wrong due to the misunderstanding. 

With the same reason why you add this back, @marmbrus 's suggestion was not removing this forever.

File: mllib/src/main/scala/org/apache/spark/ml/image/ImageSchema.scala [hunk 2]
```diff
@@ -188,4 +189,76 @@ object ImageSchema {
...
+   * @param recursive Recursive path search flag
+   * @param numPartitions Number of the DataFrame partitions,
+   *                      if omitted uses defaultParallelism instead
+   * @param dropImageFailures Drop the files that are not valid images from the result
+   * @param sampleRatio Fraction of the files loaded
+   * @return DataFrame with a single column "image" of images;
+   *         see ImageSchema for the details
+   */
+  @deprecated("use `spark.read.format(\"image\").load(path)` and this `readImages` will be " +
+    "removed in 3.1.0.", "2.4.0")
```
Comment (dongjoon-hyun) [inline mllib/src/main/scala/org/apache/spark/ml/image/ImageSchema.scala]: Ditto. Are you sure that we agree on `will be removed in 3.1.0`?

**Comment by mengxr** [comment]
I agree with @srowen that this is an API that should be removed in 3.0.

**Comment by WeichenXu123** [comment]
Discussed with @gatorsmile , agreed to remove this in 3.0. Thanks!

**Pull request closed without merging**